### ISSUE TITLE
OntModelProvider should automatically fetch and initialize ontology model for a resource

### ISSUE THREAD

**Issue by kittaakos** [crapo/sadlos2#202]
Instead of having something like this:
```
OntModel om = OntModelProvider.find(r);
if (om == null) {
  r.getResourceServiceProvider().getResourceValidator().validate(r, CheckMode.FAST_ONLY, CancelIndicator.NullImpl);
  om = OntModelProvider.find(r);
}
```

`find` should not return with a `null` if the resource is valid. We need to make sure; cyclic dependencies are not causing SOE. Also, the ontology model should not be provided if the underlying resource has errors.

**Comment by kittaakos** [comment]
Depends on #200.

**Comment by kittaakos** [comment]
Merged via #205. Closing issue.
